pygmt/tests/test_clib.py

```diff
@@ -399,6 +402,47 @@ def test_virtualfile_from_vectors():
             assert output == expected
 
 
+@pytest.mark.xfail(
+    condition=gmt_version < Version("6.1.1"),
+    reason="GMT_Put_Strings only works for GMT 6.1.1 and above",
+)
+def test_virtual_from_vectors_one_string_column():
+    "Test passing in one column with string dtype into virtual file dataset"
+    size = 5
+    x = np.arange(size, dtype=np.int32)
+    y = np.arange(size, size * 2, 1, dtype=np.int32)
+    strings = np.array(["a", "b", "c", "d", "e"], dtype=np.str)
+    with clib.Session() as lib:
+        with lib.virtualfile_from_vectors(x, y, strings) as vfile:
+            with GMTTempFile() as outfile:
+                lib.call_module("gmtinfo", f"{vfile} ->{outfile.name}")
+                output = outfile.read(keep_tabs=True)
+        bounds = "\t".join([f"<{i.min():.0f}/{i.max():.0f}>" for i in (x, y)])
```

Comment:
By the way, [`gmtinfo`](https://docs.generic-mapping-tools.org/6.1/gmtinfo) doesn't actually report anything for the text columns in the virtualfile. So at most, this test only checks that calling `put_strings` doesn't crash PyGMT.

Reply:
Why not use `lib.write_data`?

Reply:
Because it complicates the test. But yes, I could do that.

Reply:
Actually, it's hard because unlike in `test_clib_put_strings`, we don't use `dataset = lib.create_data(...)`, so there's not 'dataset' for `lib.write_data(...)` to write to.

Reply:
OK, I see. I think you can call `gmt select` instead. If you don't give any selection criteria, it should output the original input data (If I remember it correctly, file header and segment header may be discarded, but it doesn't affect the use case here). However, due to a possible bug/limitation of **select** (see https://github.com/GenericMappingTools/gmt/issues/3907), you have to give at least one option. **gmt select -Vw** should work for the test.

Reply:
Cool, I knew there was a `gmt` command for this! Using `gmt select -Vw` works, and I've changed the test so it should be all good now.

Reply:
@weiji14 See the feedback from https://github.com/GenericMappingTools/gmt/issues/3907. **gmt convert** is a better choice, although **gmt select** also works.